contract_compassion/model/contracts.py

```diff
@@ -0,0 +1,646 @@
+
+            if contract_ids and contract_id in contract_ids:
+                if len(contract_ids) == 1:
+                    wf_service.trg_validate(uid, 'account.invoice',
+                                            invoice.id, 'invoice_cancel', cr)
+                else:
+                    invoice_obj.action_cancel_draft(
+                        cr, uid, invoice.id, context)
+
+                    inv_line_ids = [
+                        invl.id for invl in invoice_lines
+                        if invl.contract_id == contract_id]
+                    invoice_line_obj.unlink(
+                        cr, uid,
+                        inv_line_ids,
+                        context)
+
+                    invoice_obj.trg_validate(uid, 'account.invoice',
+                                             invoice.id, 'invoice_open', cr)
+
+    def _is_fully_managed(self, cr, uid, ids, field_name, arg, context):
+        """Tells if the correspondent and the payer is the same person."""
+        res = dict()
+        for contract in self.browse(cr, uid, ids, context=context):
+            res[contract.id] = contract.partner_id == contract.correspondant_id
+        return res
+
+    def get_ending_reasons(self, cr, uid, context=None):
+        """Returns all the ending reasons of sponsorships"""
+        return [
+            ('1', _("Depart of child")),
+            ('2', _("Mistake from our staff")),
+            ('3', _("Death of sponsor")),
+            ('4', _("Moved to foreign country")),
+            ('5', _("Not satisfied")),
+            ('6', _("Doesn't pay")),
+            ('8', _("Personal reasons")),
+            ('9', _("Never paid")),
+            ('10', _("Subreject")),
+            ('11', _("Exchange of sponsor")),
+            ('12', _("Financial reasons")),
+            ('25', _("Not given")),
+        ]
+
+    def _get_channels(self, cr, uid, context=None):
+        """Returns the available channel through the new sponsor
+        reached Compassion.
+        """
+        return [
+            ('postal', _("By mail")),
+            ('direct', _("Direct")),
+            ('email', _("By e-mail")),
+            ('internet', _("From the website")),
+            ('phone', _("By phone")),
+            ('payment', _("Payment")),
+        ]
+
+    def _has_mandate(self, cr, uid, ids, field_name, args, context=None):
+        # Search for an existing valid mandate
+        res = dict()
+        for contract in self.browse(cr, uid, ids, context):
+            count = self.pool.get('account.banking.mandate').search(cr, uid, [
+                ('partner_id', '=', contract.partner_id.id),
+                ('state', '=', 'valid')], count=True, context=context)
+            res[contract.id] = bool(count)
+        return res
+
+    def _name_get(self, cr, uid, ids, field_name, args, context=None):
+        return {c[0]: c[1] for c in self.name_get(cr, uid, ids, context)}
+
+    ###########################

```

Comment:
You can move the fields up in the file. It should be as high as possible, just after methods used directly in the fields declaration.